### ISSUE TITLE
Documenting default parameter values?

### ISSUE THREAD

**Issue by corinwagen** [digital-chemistry-laboratory/libdlfind#19]
Hi Kjell, this is really great work—very easy to use!

It would be nice if the default values for various parameters were documented somewhere: it seems like they're set [here](https://github.com/digital-chemistry-laboratory/libdlfind/blob/main/src/main.f90#L728), but I'm not sure if those can be overridden by default arguments in other places (I don't think I have a great understanding of how DLFIND works underneath the hood, etc). 

It seems like this is a lot of work, but maybe just for important parameters like ``iopt``, ``inithessian``, ``update``, etc?

Anyhow, I appreciate this a lot.

**Comment by kjelljorner** [comment]
Hello @corinwagen! Nice that you like it! I intentionally made it a very thin wrapper that only exposes the parameters that you get to set through the normal Fortran interface. The best source that I have to the meaning of the parameters is the [manual](https://github.com/digital-chemistry-laboratory/libdlfind/blob/main/docs/documentation.pdf), as well as the comments in the [global module](https://github.com/digital-chemistry-laboratory/libdlfind/blob/7eae5d451aedf1428ef8466fbdff2be3dd7b61c6/src/dlf_global_module.f90#L132) and I didn't explore all the options myself. I unfortunately don't have any time or intention to document this further at the moment. My idea was to keep it as close to DL-FIND and possible and refer to their (sparse) documentation.

The file that you are referencing is actually not used when you run DL-FIND, it is rather a test suite that can be used to check whether the program works as intended once you compile it. See this section in the manual:

```
However, for testing purposes, a driver module (main.f90) providing some analytic energy functions is used.
```

The parameters are set using a function `dlf_get_params` that needs to be supplied externally to DL-FIND. In libdlfind, that function is here: https://github.com/digital-chemistry-laboratory/libdlfind/blob/7eae5d451aedf1428ef8466fbdff2be3dd7b61c6/libdlfind/callback.py#L109

I think the defaults are set in this subroutine: 
https://github.com/digital-chemistry-laboratory/libdlfind/blob/7eae5d451aedf1428ef8466fbdff2be3dd7b61c6/src/dl-find.f90#L134

Which additionally calls two other subroutines:
https://github.com/digital-chemistry-laboratory/libdlfind/blob/7eae5d451aedf1428ef8466fbdff2be3dd7b61c6/src/dl-find.f90#L1934
https://github.com/digital-chemistry-laboratory/libdlfind/blob/7eae5d451aedf1428ef8466fbdff2be3dd7b61c6/src/dl-find.f90#L1946

Pull request are of course always welcome 😉

**Comment by corinwagen** [comment]
Haha, point taken! Thanks for pointing me in the right direction; I'll see if I find the time in the next few weeks to document the most common options.

**Comment by kjelljorner** [comment]
No pressure 😄 Maybe you can also find some clues in the ChemShell implementation of their DL-FIND Python implementation:
https://chemshell.org/static_files/py-chemshell/manual/build/html/opt.html (you need to sign up to download their code, which is also very much hard-wired to work with PyChemShell)

**Comment by kjelljorner** [comment]
Closed by #20
